packages/blocks/src/__internal__/rich-text/rich-text.ts

```diff
@@ -87,35 +87,30 @@ export class RichText extends LitElement {
           const nextEmbedElement = nextParentElement?.closest(selector);
           const insertedString = delta.ops[1]?.insert.toString();
           if (nextEmbedElement && nextEmbedElement !== currentEmbedElement) {
-            this.quill.deleteText(
-              retain,
-              delta.ops[1]?.insert.toString().length
-            );
+            model.text?.delete(retain, delta.ops[1]?.insert.toString().length);
             // @ts-ignore
             if (!this.host.isCompositionStart) {
-              this.quill.insertText(
-                retain,
-                delta.ops[1]?.insert.toString() || ''
-              );
+              model.text?.insert(delta.ops[1]?.insert.toString() || '', retain);
             } else {
               // FIXME we must add a noon width space to fix cursor
-              this.quill.insertEmbed(retain, 'text', ' ');
+              model.text?.insert(' ', retain);
+
               this.quill.setSelection(retain + 1, 0, 'api');
```

Comment:
Is `quill.setSelection` recommended?

Reply:
I think it's good to have a better reusable `focusRichText`, its API could be like `focusRichText(id: string, index?: number, async = false)`, which could be implemented by using `quill.setSelection` for now. Updating selection state doesn't mutate model, so using quill API is fine here. And this is also required in #161 .

cc @SaikaSakura @DiamondThree